src/field/globalfield.cxx

```diff
@@ -239,8 +241,10 @@ int GlobalField2D::msg_len(int proc) const {
 
 ///////////////////////////////////////////////////////////////////////////////////////////
 
-GlobalField3D::GlobalField3D(Mesh *m, int proc) : GlobalField(m, proc, m->GlobalNx, m->GlobalNy-2*m->ystart, m->LocalNz), 
-  data_valid(false) {
+GlobalField3D::GlobalField3D(Mesh *m, int proc)
+  : GlobalField(m, proc, m->GlobalNx,
+                m->GlobalNy-mesh->numberOfYBoundaries()*2*m->ystart, m->LocalNz),
```

Comment:
I got a warning about `mesh` being uninitialised here -- I guess `m` is correct.